src/component/loader/DropZone.tsx

```diff
@@ -68,188 +50,26 @@ function DropZone(props) {
-            case FILES_TYPES.JDX:
-            case FILES_TYPES.DX: {
-              const files = await loadFiles(selectedFilesByExtensions, {
-                asBuffer: true,
-              });
-              dispatch({ type: LOAD_JCAMP_FILE, files });
-
-              break;
-            }
-            case FILES_TYPES.JDF: {
-              const files = await loadFiles(selectedFilesByExtensions, {
-                asBuffer: true,
-              });
-              dispatch({ type: LOAD_JDF_FILE, files });
-
-              break;
-            }
-            case FILES_TYPES.ZIP: {
-              const files: File[] = await loadFiles(selectedFilesByExtensions, {
-                asBuffer: true,
-              });
-              for (const zipFile of files) {
-                const unzipResult = await Zip.loadAsync(zipFile.binary);
-
-                const uniqueFileExtensions = [
-                  ...new Set(
-                    Object.values(unzipResult.files).map((file) =>
-                      getFileExtension(file.name),
-                    ),
-                  ),
-                ];
-                const isNotZip = uniqueFileExtensions.some(
-                  (ex) =>
-                    FILES_TYPES[ex.toUpperCase()] && ex !== FILES_TYPES.ZIP,
-                );
-                if (isNotZip) {
-                  const hasBruker = Object.keys(unzipResult.files).some(
-                    (path) => {
-                      return ['2rr', 'fid', '1r'].some((brukerFile) =>
-                        path.endsWith(brukerFile),
-                      );
-                    },
-                  );
-                  if (hasBruker) dispatch({ type: LOAD_ZIP_FILE, files });
-                  await loadSubFilesFromZip(
-                    Object.values(unzipResult.files),
-                    uniqueFileExtensions,
-                  );
-                } else {
-                  dispatch({ type: LOAD_ZIP_FILE, files });
-                }
-              }
-
-              break;
-            }
-            case FILES_TYPES.NMREDATA:
-              void loadFiles(selectedFilesByExtensions, {
-                asBuffer: true,
-              }).then(async (files) => {
-                for (const zipFile of files) {
-                  dispatch({
-                    type: LOAD_NMREDATA_FILE,
-                    file: zipFile,
-                  });
-                }
-              });
-              break;
-            default:
-              throw new Error(
-                'The file extension must be zip, dx, jdx, json, mol, nmredata or nmrium.',
-              );
-          }
-        } catch (e: any) {
-          dispatch({ type: SET_LOADING_FLAG, isLoading: false });
-          alert.error(e.message);
-        }
+    (files) => {
+      files.forEach((file) => {
+        Object.defineProperty(file, 'webkitRelativePath', { value: file.path });
+      });
+
+      try {
+        dispatch({ type: LOAD_DROP_FILES, files });
```

Comment:
the async that is run inside the dispatch will not be caught in this `try...catch` statement. The error logic should move inside the action dispatcher.

Reply:
I did move the try...catch block to Reducer.